lib/bundler/definition.rb

```diff
@@ -54,7 +54,8 @@ def self.build(gemfile, lockfile, unlock)
     def initialize(lockfile, dependencies, sources, unlock, ruby_version = nil, optional_groups = [])
       @unlocking = unlock == true || !unlock.empty?
 
-      @dependencies    = dependencies
+      @dependencies = dependencies
+      @dependencies_by_name = dependencies.group_by(&:name)
```

Comment:
I guess I could move this to `additional_base_requirements_for_resolve` so that we don't hit it in `Bundler.setup` ?